src/js/components/DateInput/index.d.ts

```diff
@@ -0,0 +1,24 @@
+import * as React from "react";
+
+import { ButtonProps } from "../Button";
+import { CalendarProps } from "../Calendar";
+import { DropProps } from "../Drop";
+import { MaskedInputProps } from "../MaskedInput";
+
+export interface DateInputProps {
+  buttonProps?: ButtonProps;
```

Comment:
I think you need ButtonType instead of ButtonProps. ButtonProps doesn't include the intrinsic element of Button.
This comment might be relevant for the rest of the imported props.

Reply:
yes, changed